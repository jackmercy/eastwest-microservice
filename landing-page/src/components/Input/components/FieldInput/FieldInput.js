import React, { memo } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import useStyles from './fieldInput.styles';

const FieldInput = ({
  tag: Component,
  size,
  hasError,
  className,
  ...restProps
}) => {
  const classes = useStyles();

  return (
    <Component
      className={classNames(classes.root, className, size, hasError && 'error')}
      {...restProps}
    />
  );
};

FieldInput.propTypes = {
  tag: PropTypes.node,
  hasError: PropTypes.bool,
  className: PropTypes.string,
  size: PropTypes.oneOf(['large', 'medium', 'small']),
};
FieldInput.defaultProps = {
  tag: 'input',
  size: 'medium',
  className: '',
  hasError: false,
};

export default memo(FieldInput);
