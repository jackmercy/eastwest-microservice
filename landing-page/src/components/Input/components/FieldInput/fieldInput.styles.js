import { createUseStyles } from 'react-jss';

import COLORS from '../../../../constants/COLORS';
import { hexToRgbA, rem } from '../../../../utils/jss';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    fontFamily: 'inherit',
    width: '100%',
    lineHeight: 'normal',
    fontWeight: 400,
    boxSizing: 'border-box',
    padding: [0, rem(16)],
    color: COLORS.textBlack01,
    backgroundColor: COLORS.bgWhite,
    border: `1px solid ${hexToRgbA(COLORS.borderGray02, 0.5)}`,
    borderRadius: rem(5),
    '&:focus': { outline: 'none', borderColor: COLORS.primary },
    '&::placeholder': { color: COLORS.textGray },
    '-moz-appearance': 'textfield',
    '&::-webkit-outer-spin-button, &::-webkit-inner-spin-button': {
      '-webkit-appearance': 'none',
    },
    '&::-ms-clear, &::-ms-reveal': { display: 'none' },
    '&:-ms-input-placeholder': { color: COLORS.textGray },
    '&:disabled': {
      cursor: 'not-allowed',
      color: COLORS.textGray,
      backgroundColor: COLORS.bgGray02,
    },
    '&.large': { height: rem(48), fontSize: rem(16) },
    '&.medium': { height: rem(40), fontSize: rem(16) },
    '&.small': { height: rem(32), fontSize: rem(14) },
    '&.error': { borderColor: COLORS.borderRed },
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    root: {
      '&.large, &.medium, &.small': { fontSize: rem(14), height: rem(32) },
    },
  },
});
