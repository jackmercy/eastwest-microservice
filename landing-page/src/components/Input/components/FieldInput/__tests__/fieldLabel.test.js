import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FieldInput from '../FieldInput';

describe('Testing Input component > FieldInput', () => {
  it('should render match snapshot', () => {
    const wrapper = shallow(<FieldInput name="somename" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
