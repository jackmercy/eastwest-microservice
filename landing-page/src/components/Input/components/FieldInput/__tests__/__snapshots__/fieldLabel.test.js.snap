// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input component > FieldInput should render match snapshot 1`] = `
<input
  className="root-0-2-1 medium"
  name="somename"
/>
`;
